068-kafkabrige-consumer-producer.md

```diff
@@ -0,0 +1,83 @@
+# Enhance KafkaBridge resource with consumer inactivity timeout and HTTP consumer/producer parts enablement
+
+Providing support in the Strimzi Kubernetes Operator for following properties supported in the Strimzi HTTP Kafka Bridge:
+* `http.timeoutSeconds`   - For deleting inactive consumers after a timeout (disabled by default).
+* `http.consumer.enabled` - To enable/disable the HTTP consumer part (enabled by default).
+* `http.producer.enabled` - To enable/disable the HTTP producer part (enabled by default.
+
+## Current situation
+
+Properties are not yet supported in the Strimzi Kubernetes Operator.
+
+## Motivation
+
+Raised in the discussion here: https://github.com/strimzi/strimzi-kafka-operator/issues/8732 and triaged on 29.6.2023: We should enable the configuration of these fields. But how should the API look like? Should have a proposal to clarify the API changes.
+
+## Proposal
+
+Proposal is implemented here: https://github.com/strimzi/strimzi-kafka-operator/pull/9820, both Paolo and Jakub suggested to raise a proposal for the implementation of `http.consumer.enabled` and `http.producer.enabled` specifically, as it can be done in two ways, current status is that you can just enable/disable the usage of consumer or producer on the HTTP side, which means being or not being able to use HTTP to producer and/or consume (so by using the corresponding Kafka part).
+Question is whether the enablement should be located as part of the `http` properties, or in each of the corresponding `consumer` or `producer` properties.
+
+### Version 1 (which is currently implemented):
+
+Adding dedicated section under the `http` properties for defining enablement of producer and consumer:
+```yaml
+apiVersion: "kafka.strimzi.io/v1beta2"
+kind: "KafkaBridge"
+metadata:
+  name: "test-kafka-bridge"
+spec:
+  replicas: 1
+  image: "my-test-image"
+  bootstrapServers: "my-cluster-kafka:9092"
+  http:
+    timeoutSeconds: 60
+    producer:
+      enabled: false
+    consumer:
+      enabled: true
+  consumer:
+    config:
+      foo: "bar"
+  producer:
+    config:
+      foo: "buz"
+  enableMetrics: false
+```
+
+### Version 2:
+
+Adding enablement of producer and consumer to their own sections, example:
+```yaml
+apiVersion: "kafka.strimzi.io/v1beta2"
+kind: "KafkaBridge"
+metadata:
+  name: "test-kafka-bridge"
+spec:
+  replicas: 1
+  image: "my-test-image"
+  bootstrapServers: "my-cluster-kafka:9092"
+  http:
+    timeoutSeconds: 60
+  consumer:
+    enabled: true
+    config:
+      foo: "bar"
+  producer:
+    enabled: false
+    config:
+      foo: "buz"
```

Comment:
I prefer Version 2 and it also lets me think that we made a mistake in the past to put enablement of producer and consumer under the `http.` section. In the end, producer and consumer are an intrinsic part of the bridge or anyway when enabling/disabling one of them you are actually modifying the overall behaviour for both. I mean, you are enabling/disabling both HTTP and Kafka part of each. There is no distinction.

Reply:
Maybe it should be called `httpEnabled` instead of just `enabled` then?

If you'd like, I can fix this in the operator API and then we can look at moving it inside the Bridge after?

Reply:
No I think that `enabled` is the right word to use because I don't see it totally related to HTTP protocol.
Also there are no changes to make on the PR. It has to be parked. We need to reach an agreement on the API through the proposal with all interested maintainers.
Even because just because I like Version 2, it doesn't mean we'll go with Version 2 :-) ... we need to wait what the others think.